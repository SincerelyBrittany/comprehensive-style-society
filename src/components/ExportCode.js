import React from 'react';
import '../css/ExportCode.scss';
export default class ExportCode extends React.Component {
    constructor(props) {
        super(props);
        // const [copySuccess, setCopySuccess] = useState('');
        // const copyToClipBoard = async copyMe => {
        //     try {
        //       await navigator.clipboard.writeText(copyMe);
        //       setCopySuccess('Copied!');
        //     } catch (err) {
        //       setCopySuccess('Failed to copy!');
        //     }
        // };
    }

    render() {
        var className = "myButton";
        var buttonText = "BUTTON";
        return (
            <div>
                <code id="html-code" class="language">
                    <span class="token tag">
                        <span class="token tag">
                            <span class="token punctuation">&lt;</span>
                            a
                        </span>
                        &nbsp;
                        <span class="token attr-name">href</span>
                        <span class="token attr-value">
                            <span class="token punctuation">="</span>
                            #
                            <span class="token punctuation">"</span>
                        </span>
                        &nbsp;
                        <span class="token attr-name">class</span>
                        <span class="token attr-value">
                            <span class="token punctuation">="</span>
                            {className}
                            <span class="token punctuation">"</span>
                        </span>
                        <span class="token punctuation">&gt;</span>
                    </span>
                    {buttonText}
                    <span class="token tag">
                        <span class="token tag">
                            <span class="token punctuation">&lt;/</span>
                            a
                        </span>
                        <span class="token punctuation">&gt;</span>
                    </span>
                </code>
                <br></br>
                <code id="css-code" class="language">
                    <span class="token selector">.{className} </span>
                    <span class="token punctuation">&#123;</span><br></br>

                    <span class="token property">font-size</span>
                    <span class="token punctuation">:</span>
                    &nbsp;{this.props.fontSize}
                    <span class="token punctuation">;</span>
                    
                    <br></br><span class="token punctuation">&#125;</span>
                </code>
                {/* <button onClick={this.copyToClipboard}>Copy CSS</button>
                {this.state.copySuccess} */}
            </div>
        );
    }
}