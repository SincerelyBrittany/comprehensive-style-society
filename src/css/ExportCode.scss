.token {
    color: lightgrey;
    font-family: Consolas,Monaco,'Andale Mono',monospace;
    font-size: 11px;
    text-align: left;
    word-spacing: normal;
    tab-size: 4;
    hyphens: none;
}
.token .tag {
    color: purple;
}
.token .attr-name{
    color: greenyellow;
}
.token .attr-value{
    color: lightskyblue;
}
.selector {
    color: greenyellow;
}
.property {
    color: purple;
}
.language {
    color: black;
    font-size: 11px;
}