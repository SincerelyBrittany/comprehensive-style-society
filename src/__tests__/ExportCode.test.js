import React from 'react';
import renderer from 'react-test-renderer';
import ExportCode from '../components/ExportCode';

it('will check ExportCode component renders', () => {
    const component = renderer.create(<ExportCode />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

it('will check the font-size attribute', () => {
    const props =  {
        fontSize: Math.floor(Math.random() * 20)
    }
    expect(props).toMatchSnapshot({
      fontSize: expect.any(Number),
    });
});

it('renders as an anchor when no page is set', () => {
    const tree = renderer.create(<ExportCode fontSize="16"/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
