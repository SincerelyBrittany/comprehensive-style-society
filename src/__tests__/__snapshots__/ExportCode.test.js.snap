// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as an anchor when no page is set 1`] = `
<div>
  <code
    class="language"
    id="html-code"
  >
    <span
      class="token tag"
    >
      <span
        class="token tag"
      >
        <span
          class="token punctuation"
        >
          &lt;
        </span>
        a
      </span>
       
      <span
        class="token attr-name"
      >
        href
      </span>
      <span
        class="token attr-value"
      >
        <span
          class="token punctuation"
        >
          ="
        </span>
        #
        <span
          class="token punctuation"
        >
          "
        </span>
      </span>
       
      <span
        class="token attr-name"
      >
        class
      </span>
      <span
        class="token attr-value"
      >
        <span
          class="token punctuation"
        >
          ="
        </span>
        myButton
        <span
          class="token punctuation"
        >
          "
        </span>
      </span>
      <span
        class="token punctuation"
      >
        &gt;
      </span>
    </span>
    BUTTON
    <span
      class="token tag"
    >
      <span
        class="token tag"
      >
        <span
          class="token punctuation"
        >
          &lt;/
        </span>
        a
      </span>
      <span
        class="token punctuation"
      >
        &gt;
      </span>
    </span>
  </code>
  <br />
  <code
    class="language"
    id="css-code"
  >
    <span
      class="token selector"
    >
      .
      myButton
       
    </span>
    <span
      class="token punctuation"
    >
      {
    </span>
    <br />
    <span
      class="token property"
    >
      font-size
    </span>
    <span
      class="token punctuation"
    >
      :
    </span>
     
    16
    <span
      class="token punctuation"
    >
      ;
    </span>
    <br />
    <span
      class="token punctuation"
    >
      }
    </span>
  </code>
</div>
`;

exports[`will check ExportCode component renders 1`] = `
<div>
  <code
    class="language"
    id="html-code"
  >
    <span
      class="token tag"
    >
      <span
        class="token tag"
      >
        <span
          class="token punctuation"
        >
          &lt;
        </span>
        a
      </span>
       
      <span
        class="token attr-name"
      >
        href
      </span>
      <span
        class="token attr-value"
      >
        <span
          class="token punctuation"
        >
          ="
        </span>
        #
        <span
          class="token punctuation"
        >
          "
        </span>
      </span>
       
      <span
        class="token attr-name"
      >
        class
      </span>
      <span
        class="token attr-value"
      >
        <span
          class="token punctuation"
        >
          ="
        </span>
        myButton
        <span
          class="token punctuation"
        >
          "
        </span>
      </span>
      <span
        class="token punctuation"
      >
        &gt;
      </span>
    </span>
    BUTTON
    <span
      class="token tag"
    >
      <span
        class="token tag"
      >
        <span
          class="token punctuation"
        >
          &lt;/
        </span>
        a
      </span>
      <span
        class="token punctuation"
      >
        &gt;
      </span>
    </span>
  </code>
  <br />
  <code
    class="language"
    id="css-code"
  >
    <span
      class="token selector"
    >
      .
      myButton
       
    </span>
    <span
      class="token punctuation"
    >
      {
    </span>
    <br />
    <span
      class="token property"
    >
      font-size
    </span>
    <span
      class="token punctuation"
    >
      :
    </span>
     
    <span
      class="token punctuation"
    >
      ;
    </span>
    <br />
    <span
      class="token punctuation"
    >
      }
    </span>
  </code>
</div>
`;

exports[`will check the font-size attribute 1`] = `
Object {
  "fontSize": Any<Number>,
}
`;
